File: 2023-2/Basic/gyeongtae/Deque/10866.cpp
#include <iostream>
#include <deque>

using namespace std;

int main() {
    deque <int> deq;
    int a, b;
    string n, m;
    cin >> a;
    for (int i = 0;i < a;i++) {
        cin >> n;
        if (n == "push_back") {
            cin >> b;
            deq.push_back(b);
        }
        else if (n == "push_front") {
            cin >> b;
            deq.push_front(b);
        }
        else if (n == "pop_front") {
            if (!deq.empty()) {
                cout << deq.front() << "\n";
                deq.pop_front();
            }
            else {
                cout << "-1\n";
            }
        }
        else if (n == "pop_back") {
            if (!deq.empty()) {
                cout << deq.back() << "\n";
                deq.pop_back();
            }
            else {
                cout << "-1\n";
            }
        }
        else if (n == "size") {
            cout << deq.size() << "\n";
        }
        else if (n == "empty") {
            cout << deq.empty() << "\n";
        }
        else if (n == "front") {
            if (!deq.empty()) {
                cout<<deq.front()<<"\n";
            }
            else {
                cout << "-1\n";
            }
        }
        else if (n == "back") {
            if (!deq.empty()) {
                cout << deq.back() << "\n";
            }
            else {
                cout << "-1\n";
            }
        }
    }
}

File: 2023-1/Basic/bjgood01/Deque/10866.py
import sys
from collections import deque
n = int(sys.stdin.readline().rstrip())
queue = deque()
for i in range(n):
    s = list(sys.stdin.readline().rstrip().split())
    if s[0] == 'push_front':
        queue.insert(0, int(s[1]))  # 정수를 덱의 앞에 넣는다
    elif s[0] == 'push_back':
        queue.append(int(s[1]))
    elif s[0] == 'pop_front':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.popleft())
    elif s[0] == 'pop_back':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue.pop())
    elif s[0] == 'size':
        print(len(queue))
    elif s[0] == 'empty':
        if len(queue) == 0:
            print(1)
        else:
            print(0)
    elif s[0] == 'front':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[0])
    elif s[0] == 'back':
        if len(queue) == 0:
            print(-1)
        else:
            print(queue[-1])
